src/outputs/ascent.cpp

```diff
@@ -209,16 +209,7 @@ void AscentOutput::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm,
 #endif // ifndef PARTHENON_ENABLE_ASCENT
 
   // advance output parameters
-  output_params.file_number++;
-  pin->SetInteger(output_params.block_name, "file_number", output_params.file_number);
-  if (output_params.dt > 0.0) {
-    output_params.next_time += output_params.dt;
-    pin->SetReal(output_params.block_name, "next_time", output_params.next_time);
-  }
-  if (output_params.dn > 0) {
-    output_params.next_n += output_params.dn;
-    pin->SetInteger(output_params.block_name, "next_n", output_params.next_n);
-  }
+  UpdateNextOutput_(pm, tm);
```

Comment:
Could you just call `ptype->UpdateNextOutput_(pm, tm)` after `WriteOutputFile(...)` in `Outputs::MakeOutputs` so this doesn't need to be called within every type of output?

Reply:
oh actually no, this doesn't quite work. Because different output types handle this differently, and it needs to be updated at different times, specifically for restarts.